src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/Common/AdapterUtil.SqlClient.cs

```diff
@@ -757,29 +757,37 @@ internal static Version GetAssemblyVersion()
         // This method assumes dataSource parameter is in TCP connection string format.
         internal static bool IsAzureSqlServerEndpoint(string dataSource)
         {
+            int length = dataSource.Length;
             // remove server port
-            int i = dataSource.LastIndexOf(',');
-            if (i >= 0)
+            int foundIndex = dataSource.LastIndexOf(',', length - 1, length - 1);
+            if (foundIndex >= 0)
             {
-                dataSource = dataSource.Substring(0, i);
+                length = foundIndex;
             }
 
             // check for the instance name
-            i = dataSource.LastIndexOf('\\');
-            if (i >= 0)
+            foundIndex = dataSource.LastIndexOf('\\', length - 1, length - 1);
```

Comment:
These ones are needed. if there was a comma we need to only search before the comma so that we're ignoring the remainder of the string. The original substring-ed so it didn't need to use them.